lib/matplotlib/__init__.py

```diff
@@ -1515,6 +1516,151 @@ def test(verbosity=1):
+        include all other keyword parameter. Only needed if the wrapped function does contain
+        `*args` and replace_names is not None.
+    """
+    if replace_names is not None:
+        replace_names = set(replace_names)
+
+    def param(func):
+        # this call is deprecated in 3.x and will be removed in py3.6 :-(
+        # TODO: implement the same for py 3.x/3.6+ with inspect.signature(..)
+        arg_spec = inspect.getargspec(func)
+        _arg_names = arg_spec.args
+        _has_no_varargs = arg_spec.varargs is None
+        _has_varkwargs = not arg_spec.keywords is None
+
+        # there can't be any positional arguments behind *args and no positional args can end up
+        # in **kwargs, so we only need to check for varargs:
+        # http://stupidpythonideas.blogspot.de/2013/08/arguments-and-parameters.html
+        if _has_no_varargs:
+            # remove the first "ax" arg
+            arg_names = _arg_names[1:]
+        else:
+            # in this case we need a supplied list of arguments or we need to replace all variables
+            # -> compile time check
+            if replace_names is None:
+                # all argnames should be replaced
+                arg_names = None
+            elif len(replace_names) == 0:
+                # No argnames should be replaced
+                arg_names = []
+            else:
+                assert not (positional_parameter_names is None), "Got replace_names and wrapped function uses *args, need positional_parameter_names!"
+                # remove ax arg
+                arg_names = positional_parameter_names[1:]
+
+        # compute the possible label_namer and label position in positional arguments
+        label_pos = 9999 # bigger than all "possible" argument lists
+        label_namer_pos = 9999 # bigger than all "possible" argument lists
+        if label_namer and arg_names and (label_namer in arg_names):
+            label_namer_pos = arg_names.index(label_namer)
+            if "label" in arg_names:
+                label_pos = arg_names.index("label")
+
+        # Check the case we know a label_namer but we can't find it the arg_names...
+        # Unfortunately the label_namer can be in **kwargs,  which we can't detect here and which
+        # results in a non-set label which might surprise the user :-(
+        if label_namer and not _has_varkwargs:
+            if not arg_names:
+                msg = "label_namer '%s' can't be found as the parameter without 'positional_parameter_names'."
+                raise AssertionError(msg % (label_namer))
+            elif label_namer not in arg_names:
+                msg = "label_namer '%s' can't be found in the parameter names (known argnames: %s)."
+                raise AssertionError(msg % (label_namer, arg_names))
+            else:
+                # this is the case when the name is in arg_names
+                pass
+
+        @functools.wraps(func)
```

Comment:
See https://github.com/matplotlib/matplotlib/pull/4488/files for an example how to do this in a brute-forcey kind of way.  It might be necessary to just add it by hand to everything (it's only ~50 places to paste).